aws-datastore/src/main/java/com/amplifyframework/datastore/AWSDataStorePlugin.java

```diff
@@ -98,6 +103,20 @@ private AWSDataStorePlugin(
             () -> api.getPlugins().isEmpty() ? Orchestrator.Mode.LOCAL_ONLY : Orchestrator.Mode.SYNC_VIA_API
         );
         this.userProvidedConfiguration = userProvidedConfiguration;
+
+        // Listen for API_ENDPOINT_STATUS_CHANGED to be emitted by the API and trigger
+        // the DataStore NETWORK_STATUS event accordingly.
+        Amplify.Hub.subscribe(HubChannel.API, hubEvent -> {
+            return ApiChannelEventName.API_ENDPOINT_STATUS_CHANGED.name().equals(hubEvent.getName());
+        }, hubEvent -> {
+                if (hubEvent.getData() instanceof ApiEndpointStatusChangeEvent) {
+                    ApiEndpointStatusChangeEvent eventData = (ApiEndpointStatusChangeEvent) hubEvent.getData();
+                    boolean isActive = ApiEndpointStatus.REACHABLE.equals(eventData.getCurrentStatus());
+                    Amplify.Hub.publish(HubChannel.DATASTORE,
+                        HubEvent.create(DataStoreChannelEventName.NETWORK_STATUS, new NetworkStatusEvent(isActive))
+                    );
+                }
+            });
```

Comment:
Can you shuffle this off to a helper class, and then do a thin integration to the plugin with like 
```
NetworkStatusMonitor.enable();
```
Or something like that?

Also, does this _need_ to be int the constructor? Or could it go down into the `configure(...)` or `initialize(...)` hooks? Which is the most appropriate lifecycle hook for this?

Reply:
I tried to clean up a bit, by moving the casting logic into the event types. That being said, those methods throw exceptions so I ended up with an ugly try/catch here :( 

Maybe I'll just encapsulate the whole subscribe block somewhere. 

I think I can probably move this out of the constructor. It just has to happen before we trigger the sync.

Reply:
Ok...I move it out to configure and did a variation of what you suggested with regards to encapsulating that block of code to listen and trigger events.